### PR TITLE
Configure Alertmanager and add capacity related rules for openebs pools and volumes 

### PR DESCRIPTION
**Pull request by rajaSahil**
### What this PR does?
- Configure Alertmanager
- Add rules for openebs pools and volumes(Only capacity related rules)
- Automate loading of rules
- Enabled/Disable rules.

### CODE REVIEW

File: deploy/charts/openebs-monitoring/README.md [hunk 1]
```diff
@@ -87,24 +87,31 @@ helm install openebs-monitoring openebs-monitoring/monitoring --namespace openeb
 | ----------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------- | --------------------------------------------------------- |
 | `kube-prometheus-stack.prometheus.service.type`                                           | Service type for Prometheus                                                         | `"NodePort"`                                              |
 | `kube-prometheus-stack.prometheus.service.nodePort`                                       | NodePort value for Prometheus service                                               | `32514`                                                   |
-| `kube-prometheus-stack.grafana.service.type`                                              | Service type for Grafana                                                            | `"NodePort"`                                              |
```
Comment (kmova) [inline deploy/charts/openebs-monitoring/README.md:90]: q_: why are these removed?

File: deploy/charts/openebs-monitoring/values.yaml [hunk 2]
```diff
@@ -120,6 +144,12 @@ kube-prometheus-stack:
       - grafana-polystat-panel
       - snuids-trafficlights-panel
 
+customRules:
```
Comment (kmova) [inline deploy/charts/openebs-monitoring/values.yaml]: n_: call this alertrules?

File: deploy/charts/openebs-monitoring/README.md (hunk 1 above)
Comment (rajaSahil) [reply deploy/charts/openebs-monitoring/README.md:90]: I have moved it below.

File: deploy/charts/openebs-monitoring/values.yaml [hunk 3]
```diff
@@ -120,6 +144,12 @@ kube-prometheus-stack:
       - grafana-polystat-panel
       - snuids-trafficlights-panel
 
+alertRules:
+  pools:
```
Comment (Ab-hishek) [inline deploy/charts/openebs-monitoring/values.yaml]: Instead of pools can it be `cstorPools`? This way readability will increase.

File: deploy/charts/openebs-monitoring/values.yaml [hunk 4]
```diff
@@ -120,6 +144,12 @@ kube-prometheus-stack:
...
+    enabled: true
+  volumes:
```
Comment (Ab-hishek) [inline deploy/charts/openebs-monitoring/values.yaml]: Same comment as above for `volumes` also if it is specific to cstor volumes only.

Comment (rajaSahil) [reply deploy/charts/openebs-monitoring/values.yaml]: It is not specific to cStor volumes.

File: deploy/charts/openebs-monitoring/values.yaml (hunk 3 above)
Comment (rajaSahil) [reply deploy/charts/openebs-monitoring/values.yaml]: Since we are following the naming convention of pools and volumes, I think it is good.

File: deploy/charts/openebs-monitoring/values.yaml (hunk 4 above)
Comment (kmova) [reply deploy/charts/openebs-monitoring/values.yaml]: if volumes and pools are using standard metrics - then we can keep them generic. For example when Mayastor or LVM rules are added, will the above still apply.

Comment (kmova) [reply deploy/charts/openebs-monitoring/values.yaml]: If the answer to the above - is not clear or "no", then it makes sense to keep engine-specific rules under engine -- like cStorPool and cStorVolume.

File: deploy/charts/openebs-monitoring/README.md [hunk 5]
```diff
@@ -87,24 +87,31 @@ helm install openebs-monitoring openebs-monitoring/monitoring --namespace openeb
...
 | `kube-prometheus-stack.prometheus.prometheusSpec.storageSpec`                             | Storage spec to specify how storage shall be used.                                  | `{}`                                                      |
 | `kube-prometheus-stack.alertmanager.enabled`                                              | Deploy alertmanager                                                                 | `true`                                                    |
+| `kube-prometheus-stack.alertmanager.config`                                              | Provide YAML to configure Alertmanager.                                                                 | `{global: {resolve_timeout: 5m}, route: {group_by: [alertname, job, volName], group_wait: 30s, group_interval: 5m, repeat_interval: 4h, receiver: 'null'}, receivers: [{name: 'null'}], templates: [/etc/alertmanager/config/*.tmpl]}`                                                    |
```
Comment (Ab-hishek) [inline deploy/charts/openebs-monitoring/README.md]: Probably here, we can follow the convention of specifying default values as `{...}` as is done for relabelings.

**Review by Ab-hishek — APPROVED** [review]

File: deploy/charts/openebs-monitoring/rules/Jiva/jiva-volume-rules.yaml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+          openebsVolume: "{{ $labels.openebs_pv }}"
+          openebsVolumeClaim: "{{ $labels.openebs_pvc }}"
+          description: |-
+            OpenEBS volume '{{$labels.openebs_pv}}' created for '{{$labels.openebs_pvc}}' is unreachable since last 2 minutes and it needs to be brought up
+          summary: "OpenEBS volume '{{$labels.openebs_pv}}' created for the claim '{{$labels.openebs_pvc}}' is unreachable since last 2 minutes."
+        expr: |-
+          ((openebs_volume_uptime{openebs_jiva_label="jiva-controller"}) and (openebs_volume_uptime{openebs_jiva_label="jiva-controller"} > openebs_volume_uptime{openebs_jiva_label="jiva-controller"} offset 2m)) unless ((openebs_volume_uptime{openebs_jiva_label="jiva-controller"} offset 4m) and (openebs_volume_uptime{openebs_jiva_label="jiva-controller"} offset 2m > openebs_volume_uptime{openebs_jiva_label="jiva-controller"} offset 4m))
+        for: 2m
+        labels:
+          severity: "info"
```
Comment (kmova) [inline deploy/charts/openebs-monitoring/rules/Jiva/jiva-volume-rules.yaml]: should this be warning?

File: deploy/charts/openebs-monitoring/rules/Jiva/jiva-volume-rules.yaml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      - alert: VolumeDegraded
+        annotations:
+          componentType: "volume"
+          openebsVolume: "{{ $labels.openebs_pv }}"
+          openebsVolumeClaim: "{{ $labels.openebs_pvc }}"
+          description: |-
+            OpenEBS volume '{{$labels.openebs_pv}}' created for the claim '{{$labels.openebs_pvc}}' is degraded, check the replica statuses
+          summary: "OpenEBS volume '{{$labels.openebs_pv}}' created for the claim '{{$labels.openebs_pvc}}' is in a degraded state"
+        expr: |-
+          (openebs_volume_status{openebs_jiva_label="jiva-controller"} == 2 and (openebs_volume_status{openebs_jiva_label="jiva-controller"} offset 1m != 2))
```
Comment (kmova) [inline deploy/charts/openebs-monitoring/rules/Jiva/jiva-volume-rules.yaml]: what if the jiva is configured with single replica?

File: deploy/charts/openebs-monitoring/rules/cStor/cstor-pool-rules.yaml [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+      - alert: PoolIsDegraded
+        annotations:
+          componentType: "pool"
+          cstorPool: "{{ $labels.cstor_pool }}"
+          storageClassClaim: "{{ $labels.storage_pool_claim }}"
+          description: |-
+            cStor pool '{{$labels.cstor_pool}}' created for the pool claim '{{$labels.storage_pool_claim}}' is in a degraded state
+          summary: |-
+            cStor pool '{{$labels.cstor_pool}}' created for the pool claim '{{$labels.storage_pool_claim}}' is now online
```
Comment (kmova) [inline deploy/charts/openebs-monitoring/rules/cStor/cstor-pool-rules.yaml]: this line should probably say degraded?

**Comment by kmova** [comment]
@rajaSahil - I am wondering if it would be better to break this PR into multiple different PRs. For example, 

this PR can introduce the flags to enable alert rules and include only volume capacity and pool capacity alerts that can be easily tested. 

The follow-up PRs can add one alert at a time, with details on how the alert was tested.

**Review by kmova — APPROVED** [review]
